src/index.ts

```diff
@@ -1,3 +1,182 @@
 // Copyright (c) Jupyter Development Team.
 // Distributed under the terms of the Modified BSD License.
 'use-strict';
+
+import {
+  IInputAreaViewModel
+} from 'jupyter-js-input-area';
+
+import {
+  IOutputAreaViewModel
+} from 'jupyter-js-output-area';
+
+import {
+  IObservableList
+} from 'phosphor-observablelist';
+
+import {
+  Widget
+} from 'phosphor-widget';
+
+import './index.css';
+
+
+/**
+ * An enum which describes the type of cell.
+ */
+enum CellType {
+  /**
+   * The cell contains code input.
+   */
+  Code,
+
+  /**
+   * The cell contains markdown.
+   */
+  Markdown,
+
+  /**
+   * The cell contains raw text.
+   */
+  Raw
+}
+
+
+/**
+ * An enum which describes which interface property has changed.
+ */
+export
+enum CellProperty {
+  CellType,
+  Tags,
+  Deletable,
+  Mergable,
+  Splittable,
+  Rendered,
+  Marked,
+  Format
+}
+
+
+/**
+ * The arguments object emitted with the `stateChanged` signal.
+ */
+export
+interface ICellChangedArgs<T> {
+  which: CellProperty,
+  oldValue: T;
+  newValue: T;
+}
+
+
+/**
+ * The definition of a model object for a base cell.
+ */
+interface IBaseCellViewModel {
+
+  /**
+   * The type of cell.
+   */
+  cellType: CellType;
+
+  /**
+   * Tags applied to the cell.
+   */
+  tags?: IObservableList<string>;
+
+  /**
+   * Get namespaced metadata about the cell.
+   */
+  getMetadata(namespace: string) : IObservableMap<string, JSONData>;
```

Comment:
Hmm, no I think it should be there still on both the cells and notebook.
Don't remember it going away. Views will definitely end up needing it.

On Tue, Nov 3, 2015 at 7:56 PM, S. Chris Colbert <notifications@github.com>
wrote:

> In src/index.ts
> <https://github.com/jupyter/jupyter-js-cells/pull/2#discussion_r43830187>:
>
> > +interface IBaseCellViewModel {
> > +
> > +  /**
> > +   * The type of cell.
> > +   */
> > +  cellType: CellType;
> > +
> > +  /**
> > +   * Tags applied to the cell.
> > +   */
> > +  tags?: IObservableList<string>;
> > +
> > +  /**
> > +   * Get namespaced metadata about the cell.
> > +   */
> > +  getMetadata(namespace: string) : IObservableMap<string, JSONData>;
>
> aren't we getting rid of metadata on the view models anyway? (we got rid
> of it on the notebook view model).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jupyter/jupyter-js-cells/pull/2/files#r43830187>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com